tools/sdk-testgen/packages/autorest.gotest/doc/gotest_howto.md

```diff
@@ -0,0 +1,156 @@
+# How to Use @autorest/gotest
+
+For Azure SDK for Go, we introduced `@autorest/gotest`, an autorest extension to auto generate the examples, tests and samples code for management modules. In this document, we will give you detailed instructions about tool's usage.
+
+## Prerequisites
+
+- [Go 1.18+](https://go.dev/dl/) is required because we leverage Go generic in our SDK.
+- Add your default `GOPATH/bin` (default is `~/go/bin`) to your system `PATH` environment.
+- Prepare local workspace of [Azure SDK for Go](https://github.com/Azure/azure-sdk-for-go) repository.  (We will use `<sdk-repo-workspace>` for SDK workspace in this document.)
+- Prepare local workspace of service swagger files. (We will use a test signalR [swagger](https://github.com/Azure/azure-sdk-tools/tree/main/tools/sdk-testgen/swagger/specification/signalr/resource-manager) as example in this document. The workspace will be `<swagger-repo-workspace>`.)
+- [Node 1.14+](https://nodejs.org/en/download/) is required and [Autorest CLI](https://github.com/Azure/autorest/tree/main/packages/apps/autorest) should be installed globally.
+
+## Config swagger
+
+We use `autorest.md` file under SDK module folder root (`<sdk-repo-workspace>/sdk/resourcemanager/signalr/armsignalr/autorest.md`) to config the service swagger file to be used for code generation. Change the file content as follows.
+
+````md
+### AutoRest Configuration
+
+> see https://aka.ms/autorest
+
+```yaml
+azure-arm: true
+require:
+- <swagger-repo-workspace>/tools/sdk-testgen/swagger/specification/signalr/resource-manager/readme.md
+- <swagger-repo-workspace>/tools/sdk-testgen/swagger/specification/signalr/resource-manager/readme.go.md
+license-header: MICROSOFT_MIT_NO_VERSION
+module-version: 0.5.0
+```
+````
+
+
+## Generate SDK
+
+After you set the swagger config, you need to regenerate the SDK first. (You could use `--generate-sdk` to generate SDK along with example/test/sample generation to skip this step.)
+
+1. Delete all `.go` files under the module folder.

```

Comment:
Is it possible to replace the existing file for autorest instead of delete .go files?

Reply:
Is generate SDK a required step for sample generation?  
It is better to describe generate SDK with examples & add examples to the exsiting SDK.

Reply:
As the SDK existed API version will differ from the config you use when test generation, there exists a deletion process. Also, the SDK generation part is used to prevent the difference of SDK and examples/samples API version.